rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -88,6 +88,19 @@ class Publisher : public PublisherBase
       this->add_event_handler(
         options_.event_callbacks.incompatible_qos_callback,
         RCL_PUBLISHER_OFFERED_INCOMPATIBLE_QOS);
+    } else {
+      // Register default callback when not specified
+      try {
+        this->add_event_handler(
+          QOSOfferedIncompatibleQoSCallbackType(default_incompatible_qos_callback),
+          RCL_PUBLISHER_OFFERED_INCOMPATIBLE_QOS);
+      } catch (UnsupportedEventTypeException & /*exc*/) {
+        RCLCPP_WARN_ONCE(
+          rclcpp::get_logger("rclcpp::Publisher"),
```

Comment:
This should use the node's logger no?

Reply:
I am unsure how to get the node's logger from a `Publisher`/`Subscription` class. The `Publisher` and `Subscription` only has access to the `NodeBaseInterface`. How do you get access to the `NodeLoggingInterface`?

Reply:
I think that adding the logging interface to the publisher can be done in a follow-up.
For example, we're doing this currently in the code: https://github.com/ros2/rclcpp/blob/57fddca648a2c0cc422717bc2360bef2d0d90b77/rclcpp/src/rclcpp/publisher_base.cpp#L103.

Reply:
> I think that adding the logging interface to the publisher can be done in a follow-up.

I actually disagree, it will require an API change for the publisher class if we now want to pass in the logging interface, which I think we should. Better to do that now...

----

If not, you can copy what the `NodeLogger` does in the meantime:

https://github.com/ros2/rclcpp/blob/c1a7a655374a8ed68ca3f3de54adc4875c5e885d/rclcpp/src/rclcpp/node_interfaces/node_logging.cpp#L38

Reply:
Done.